src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/BasePropertyExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Linq;
+using Microsoft.Build.Framework.XamlTypes;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties
+{
+    /// <summary>
+    ///     Provides extension methods for <see cref="BaseProperty"/> instances.
+    /// </summary>
+    internal static class BasePropertyExtensions
+    {
+        /// <summary>
+        ///     Returns the value of the metadata item identified by <paramref name="metadataName"/>.
+        /// </summary>
+        /// <param name="property">The property to examine.</param>
+        /// <param name="metadataName">The name of the metadata item to return.</param>
+        /// <returns>
+        ///     The value of the corresponding metadata item, or <see langword="null"/> if it is not
+        ///     found.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        ///     <paramref name="property"/> is <see langword="null"/>.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///     <paramref name="property"/> is <see langword="null"/> or empty.
+        /// </exception>
+        public static string? GetMetadataValue(this BaseProperty property, string metadataName)
```

Comment:
GetMetadataValueOrNull. Metadata methods almost everywhere else return empty string to match MSBuild.